### PR TITLE
3.0 admin table escape html

### PR DESCRIPTION
**Pull request by nfourtythree**

### CODE REVIEW

File: src/controllers/CustomersController.php [hunk 1]
```diff
@@ -204,16 +204,16 @@ public function actionCustomersTable(): Response
 
             $rows[] = [
                 'id' => $customer['id'],
-                'title' => $customer['email'],
+                'title' => htmlspecialchars($customer['email']),
```
Comment (angrybrad) [inline src/controllers/CustomersController.php]: Everywhere you're using `htmlspecialchars`, it should be swapped to `craft\helpers\Html::encode()`

File: src/controllers/OrdersController.php [hunk 2]
```diff
@@ -1151,26 +1151,26 @@ private function _getTransactionsWIthLevelsTableArray($transactions, $level = 0)
...
+                        ['label' => htmlspecialchars(Plugin::t('Gateway Reference')), 'type' => 'code', 'value' => $transaction->reference ],
+                        ['label' => htmlspecialchars(Plugin::t('Gateway Message')), 'type' => 'text', 'value' => $transaction->message ],
+                        ['label' => htmlspecialchars(Plugin::t('Note')), 'type' => 'text', 'value' => $transaction->note ?? '' ],
+                        ['label' => htmlspecialchars(Plugin::t('Gateway Code')), 'type' => 'code', 'value' => $transaction->code ],
+                        ['label' => htmlspecialchars(Plugin::t('Converted Price')), 'type' => 'text', 'value' => Plugin::getInstance()->getPaymentCurrencies()->convert($transaction->paymentAmount, $transaction->paymentCurrency) . ' <small class="light">(' . $transaction->currency . ')</small>' . ' <small class="light">(1 ' . $transaction->currency . ' = ' . number_format($transaction->paymentRate) . ' ' . $transaction->paymentCurrency . ')</small>' ],
+                        ['label' =>htmlspecialchars (Plugin::t('Gateway Response')), 'type' => 'response', 'value' => $transaction->response ],
```
Comment (angrybrad) [inline src/controllers/OrdersController.php]: none of these are necessary - all static strings

File: src/controllers/OrdersController.php [hunk 3]
```diff
@@ -1151,26 +1151,26 @@ private function _getTransactionsWIthLevelsTableArray($transactions, $level = 0)
...
                     ],
                     'status' => [
                         'key' => $transaction->status,
-                        'label' => Plugin::t(StringHelper::toTitleCase($transaction->status))
+                        'label' => htmlspecialchars(Plugin::t(StringHelper::toTitleCase($transaction->status)))
                     ],
                     'amount' => Plugin::getInstance()->getPaymentCurrencies()->convert($transaction->amount, $transaction->currency) . ' <small class="light">(' . $transaction->currency . ')</small>',
                     'paymentAmount' => Plugin::getInstance()->getPaymentCurrencies()->convert($transaction->paymentAmount, $transaction->paymentCurrency) . ' <small class="light">(' . $transaction->currency . ')</small>',
-                    'gateway' => $transaction->gateway->name ?? Plugin::t('Missing Gateway'),
+                    'gateway' => htmlspecialchars($transaction->gateway->name ?? Plugin::t('Missing Gateway')),
```
Comment (angrybrad) [inline src/controllers/OrdersController.php]: these probably aren't necessary? all coming back from a gateway and i'm assuming a user never has a change to change them?

File: src/templates/customers/_includes/_customerTables.html [hunk 4]
```diff
@@ -48,10 +48,10 @@
     id: address.id,
     title: address.address1,
     url: address.cpEditUrl ~ (addressRedirect is defined and addressRedirect ? '?redirect=' ~ addressRedirect : ''),
-    zipCode: address.zipCode,
+    zipCode: address.zipCode|e,
     billing: (customer.primaryBillingAddressId == address.id),
     shipping: (customer.primaryShippingAddressId == address.id),
-    detail: { handle: tag('span', { 'data-icon': 'info', title: 'Show full address'|t('commerce') }) , content: addressLines|join('<br>') }
+    detail: { handle: tag('span', { 'data-icon': 'info', title: 'Show full address'|t('commerce')|e }) , content: addressLines|e|join('<br>') }
```
Comment (angrybrad) [inline src/templates/customers/_includes/_customerTables.html]: not needed on "Show full address"

File: src/templates/customers/_includes/_customerTables.html [hunk 5]
```diff
@@ -85,7 +85,7 @@ <h2>{{ 'Inactive Carts'|t('commerce') }}</h2>
   {% for order in inactiveCarts %}
     {% set inactiveTableData = inactiveTableData|merge([{
       id: order.id,
-      title: order.reference ?? order.shortNumber,
+      title: order.reference|e ?? order.shortNumber|e,
```
Comment (angrybrad) [inline src/templates/customers/_includes/_customerTables.html]: don't think a user ever has a chance to edit reference or short number

File: src/templates/customers/_includes/_customerTables.html [hunk 6]
```diff
@@ -114,7 +114,7 @@ <h2>{{ 'Subscriptions'|t('commerce') }}</h2>
       subDate: subscription.dateCreated|date('D jS M Y'),
       canDate: subscription.dateCanceled ? subscription.dateCanceled|date('D jS M Y') : '',
       expDate: subscription.dateExpired ? subscription.dateExpired|date('D jS M Y') : '',
-      active: subscription.status == 'active' ? 'Yes'|t('commerce') : 'No'|t('commerce'),
+      active: subscription.status == 'active' ? 'Yes'|t('commerce')|e : 'No'|t('commerce')|e,
```
Comment (angrybrad) [inline src/templates/customers/_includes/_customerTables.html]: not needed

File: src/templates/customers/_includes/_customerTables.html [hunk 7]
```diff
@@ -131,14 +131,14 @@ <h2>{{ 'Subscriptions'|t('commerce') }}</h2>
...
-            return '<span data-icon="check"></span>';
+            return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
         } },
     { name: 'shipping', title: Craft.t('commerce', 'Primary Shipping Address'), callback: function(value) {
             if (!value) {
                 return '';
             }
 
-            return '<span data-icon="check"></span>';
+            return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/customers/_includes/_customerTables.html]: not needed

File: src/templates/customers/_includes/_customerTables.html [hunk 8]
```diff
@@ -131,14 +131,14 @@ <h2>{{ 'Subscriptions'|t('commerce') }}</h2>
...
```
Comment (angrybrad) [inline src/templates/customers/_includes/_customerTables.html]: not needed

File: src/templates/promotions/discounts/index.html [hunk 9]
```diff
@@ -26,7 +26,7 @@
 {% for discount in discounts %}
     {% set tableData = tableData|merge([{
         id: discount.id,
-        title: discount.name,
+        title: discount.name|t('site'),
```
Comment (angrybrad) [inline src/templates/promotions/discounts/index.html]: did you mean `|e` here?

File: src/templates/promotions/discounts/index.html [hunk 10]
```diff
@@ -70,7 +70,7 @@
         { name: 'stop', title: Craft.t('commerce', 'Stops Processing?'),
             callback: function(value) {
                 if (value) {
-                    return '<span data-icon="check" title="'+Craft.t('app', 'Yes')+'"></span>';
+                    return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/promotions/discounts/index.html]: not needed

File: src/templates/promotions/discounts/index.html [hunk 11]
```diff
@@ -79,7 +79,7 @@
         { name: 'ignore', title: Craft.t('commerce', 'Ignore Sales?'),
             callback: function(value) {
                 if (value) {
-                    return '<span data-icon="check" title="'+Craft.t('app', 'Yes')+'"></span>';
+                    return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/promotions/discounts/index.html]: not needed

File: src/templates/promotions/sales/index.html [hunk 12]
```diff
@@ -23,14 +23,14 @@
...
+    sale.applyAmountAsPercent ~ ' ' ~ (sale.apply == 'byPercent' ? 'off original price'|t('commerce')|e : 'of original price'|t('commerce')|e )
     :
-    sale.applyAmountAsFlat|commerceCurrency(craft.commerce.paymentCurrencies.primaryPaymentCurrency.iso,convert=true) ~ ' ' ~ (sale.apply == 'byFlat' ? 'off original price'|t('commerce') : '(new price)'|t('commerce') )
+    sale.applyAmountAsFlat|commerceCurrency(craft.commerce.paymentCurrencies.primaryPaymentCurrency.iso,convert=true) ~ ' ' ~ (sale.apply == 'byFlat' ? 'off original price'|t('commerce')|e : '(new price)'|t('commerce')|e )
   %}
 
   {% set tableData = tableData|merge([{
     id: sale.id,
-    title: sale.name,
+    title: sale.name|t('site'),
```
Comment (angrybrad) [inline src/templates/promotions/sales/index.html]: did you mean `|e` here?

File: src/templates/promotions/sales/index.html [hunk 13]
```diff
@@ -71,7 +71,7 @@
     { name: 'ignore', title: Craft.t('commerce', 'Overrides previous?'),
         callback: function(value) {
             if (value) {
-                return '<span data-icon="check" title="'+Craft.t('app', 'Yes')+'"></span>';
+                return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/promotions/sales/index.html]: not needed

File: src/templates/promotions/sales/index.html [hunk 14]
```diff
@@ -80,7 +80,7 @@
     { name: 'stop', title: Craft.t('commerce', 'Stops subsequent?'),
         callback: function(value) {
             if (value) {
-                return '<span data-icon="check" title="'+Craft.t('app', 'Yes')+'"></span>';
+                return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/promotions/sales/index.html]: not needed

File: src/templates/settings/emails/index.html [hunk 15]
```diff
@@ -20,13 +20,13 @@
 {% for email in emails %}
     {% set tableData = tableData|merge([{
         id: email.id,
-        title: email.name,
+        title: email.name|t('site'),
```
Comment (angrybrad) [inline src/templates/settings/emails/index.html]: `|e`?

File: src/templates/settings/emails/index.html [hunk 16]
```diff
@@ -20,13 +20,13 @@
...
         status: email.enabled ? true : false,
         url: url('commerce/settings/emails/'~email.id),
-        subject: email.subject,
-        to: email.recipientType == 'custom' ? email.to : 'Customer'|t('commmerce'),
-        bcc: email.bcc,
-        template: email.templatePath,
+        subject: email.subject|t('site')|e,
+        to: email.recipientType == 'custom' ? email.to|e : 'Customer'|t('commmerce')|e,
```
Comment (angrybrad) [inline src/templates/settings/emails/index.html]: not needed on `'Customer'`

File: src/templates/settings/gateways/index.html [hunk 17]
```diff
@@ -25,9 +25,9 @@
 {% for gateway in gateways %}
     {% set tableData = tableData|merge([{
         id: gateway.id,
-        title: gateway.name,
+        title: gateway.name|t('site'),
```
Comment (angrybrad) [inline src/templates/settings/gateways/index.html]: `|e`?

File: src/templates/settings/gateways/index.html [hunk 18]
```diff
@@ -57,7 +57,7 @@
             title: Craft.t('commerce', 'Customer Enabled?'),
             callback: function(value) {
                 if (value) {
-                    return '<div data-icon="check"></div>';
+                    return '<div data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></div>';
```
Comment (angrybrad) [inline src/templates/settings/gateways/index.html]: not needed

File: src/templates/settings/lineitemstatuses/index.html [hunk 19]
```diff
@@ -26,9 +28,9 @@ <h2>{{ "Line Item Statuses"|t('commerce') }}</h2>
             url: lineItemStatus.cpEditUrl,
             html: lineItemStatus.labelHtml|raw
         },
-        title: lineItemStatus.name,
+        title: lineItemStatus.name|t('site'),
```
Comment (angrybrad) [inline src/templates/settings/lineitemstatuses/index.html]: `|e`?

File: src/templates/settings/lineitemstatuses/index.html [hunk 20]
```diff
@@ -48,7 +50,7 @@ <h2>{{ "Line Item Statuses"|t('commerce') }}</h2>
         title: Craft.t('commerce', 'Default Status?'),
         callback: function(value) {
             if (value) {
-                return '<span data-icon="check"></span>';
+                return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/settings/lineitemstatuses/index.html]: not needed

File: src/templates/settings/orderstatuses/index.html [hunk 21]
```diff
@@ -28,9 +28,9 @@ <h2>{{ "Order Statuses"|t('commerce') }}</h2>
             url: orderStatus.cpEditUrl,
             html: orderStatus.labelHtml|raw
         },
-        title: orderStatus.name,
+        title: orderStatus.name|t('site'),
```
Comment (angrybrad) [inline src/templates/settings/orderstatuses/index.html]: `|e`?

File: src/templates/settings/orderstatuses/index.html [hunk 22]
```diff
@@ -53,7 +53,7 @@ <h2>{{ "Order Statuses"|t('commerce') }}</h2>
             title: Craft.t('commerce', 'Default Status?'),
             callback: function(value) {
                 if (value) {
-                    return '<span data-icon="check"></span>';
+                    return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/settings/orderstatuses/index.html]: not needed

File: src/templates/settings/producttypes/index.html [hunk 23]
```diff
@@ -20,21 +20,21 @@
...
     {% for cat in type.getTaxCategories() %}
-    	{% set tax = tax ~ '<li>' ~ (currentUser.can('commerce-manageTax') ? '<a href="' ~ cat.getCpEditUrl() ~ '">' : '') ~ cat.name ~ (currentUser.can('commerce-manageShipping') ? '<a href="' ~ cat.getCpEditUrl() ~ '">' : '') ~'</li>' %}
+    	{% set tax = tax ~ '<li>' ~ (currentUser.can('commerce-manageTax') ? '<a href="' ~ cat.getCpEditUrl() ~ '">' : '') ~ cat.name|t('site')|e ~ (currentUser.can('commerce-manageShipping') ? '<a href="' ~ cat.getCpEditUrl() ~ '">' : '') ~'</li>' %}
     {% endfor %}
     {% set tax = tax ? tag('ul', { html: tax }) : '' %}
 
     {% set tableData = tableData|merge([{
         id: type.id,
-        title: type.name,
+        title: type.name|t('site'),
```
Comment (angrybrad) [inline src/templates/settings/producttypes/index.html]: `|e`?

File: src/templates/settings/producttypes/index.html [hunk 24]
```diff
@@ -48,7 +48,7 @@
         { name: 'hasVariants', title: Craft.t('commerce', 'Has Variants?'),
             callback: function(value) {
                 if (value) {
-                    return '<span data-icon="check"></span>';
+                    return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/settings/producttypes/index.html]: not needed

File: src/templates/shipping/shippingcategories/index.html [hunk 25]
```diff
@@ -14,10 +14,10 @@
 {% for shippingCategory in shippingCategories %}
     {% set tableData = tableData|merge([{
         id: shippingCategory.id,
-        title: shippingCategory.name,
+        title: shippingCategory.name|t('site'),
```
Comment (angrybrad) [inline src/templates/shipping/shippingcategories/index.html]: `|e`?

File: src/templates/shipping/shippingcategories/index.html [hunk 26]
```diff
@@ -32,7 +32,7 @@
             title: Craft.t('app', 'Default'),
             callback: function(value) {
                 if (value) {
-                    return '<div data-icon="check"></div>';
+                    return '<div data-icon="check" title="'+Craft.escapeHtml(Craft.t('app','Yes'))+'"></div>';
```
Comment (angrybrad) [inline src/templates/shipping/shippingcategories/index.html]: not needed

File: src/templates/shipping/shippingzones/index.html [hunk 27]
```diff
@@ -20,11 +20,11 @@
 {% for shippingZone in shippingZones %}
 	{% set tableData = tableData|merge([{
         id: shippingZone.id,
-        title: shippingZone.name,
+        title: shippingZone.name|t('site'),
```
Comment (angrybrad) [inline src/templates/shipping/shippingzones/index.html]: `|e`?

File: src/templates/shipping/shippingzones/index.html [hunk 28]
```diff
@@ -20,11 +20,11 @@
...
         url: url('commerce/shipping/shippingzones/'~shippingZone.id),
-        description: shippingZone.description,
-        type: shippingZone.isCountryBased ? 'Country-based'|t('commerce') : 'State-based'|t('commerce'),
-        countryState: shippingZone.isCountryBased ? shippingZone.getCountriesNames()|join(', ') : shippingZone.getStatesNames()|join(', '),
+        description: shippingZone.description|t('site')|e,
+        type: shippingZone.isCountryBased ? 'Country-based'|t('commerce')|e : 'State-based'|t('commerce')|e,
```
Comment (angrybrad) [inline src/templates/shipping/shippingzones/index.html]: not needed

File: src/templates/store-settings/countries/index.html [hunk 29]
```diff
@@ -19,10 +23,10 @@
 {% for country in countries %}
     {% set tableData = tableData|merge([{
         id: country.id,
-        title: country.name,
+        title: country.name|t('site'),
```
Comment (angrybrad) [inline src/templates/store-settings/countries/index.html]: `|e`?

File: src/templates/store-settings/countries/index.html [hunk 30]
```diff
@@ -19,10 +23,10 @@
...
         status: country.enabled ? true : false,
         url: url('commerce/store-settings/countries/'~country.id),
-        countryCode: country.iso,
+        countryCode: country.iso|e,
```
Comment (angrybrad) [inline src/templates/store-settings/countries/index.html]: is this user definable?

File: src/templates/store-settings/countries/index.html [hunk 31]
```diff
@@ -35,7 +39,7 @@
         { name: 'stateRequired', title: Craft.t('commerce', 'State Required?'),
             callback: function(value) {
                 if (value) {
-                    return '<span data-icon="check" title="'+Craft.t('app', 'Yes')+'"></span>';
+                    return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/store-settings/countries/index.html]: not needed

File: src/templates/store-settings/paymentcurrencies/index.html [hunk 32]
```diff
@@ -21,7 +21,7 @@
 {% for currency in currencies %}
 	{% set tableData = tableData|merge([{
         id: currency.id,
-        title: currency.primary ? "{name} (Primary)"|t('commerce', { name: currency.name }) : currency.name,
+        title: currency.primary ? "{name} (Primary)"|t('commerce', { name: currency.name })|e : currency.name|t('site'),
```
Comment (angrybrad) [inline src/templates/store-settings/paymentcurrencies/index.html]: `|e` for `currency.name` if it's user definable?

File: src/templates/store-settings/paymentcurrencies/index.html [hunk 33]
```diff
@@ -38,7 +38,7 @@
             title: Craft.t('commerce', 'Conversion Rate'),
             callback: function(value) {
                 if (value == 'base') {
-                    return '<span class="token">'+Craft.t('commerce', 'Base')+'</span>'
+                    return '<span class="token">'+Craft.escapeHtml(Craft.t('commerce', 'Base'))+'</span>'
```
Comment (angrybrad) [inline src/templates/store-settings/paymentcurrencies/index.html]: not needed

File: src/templates/store-settings/states/index.html [hunk 34]
```diff
@@ -18,13 +21,13 @@
 {% for state in states %}
 	{% set tableData = tableData|merge([{
         id: state.id,
-        title: state.name,
+        title: state.name|t('site'),
```
Comment (angrybrad) [inline src/templates/store-settings/states/index.html]: `|e`?

File: src/templates/tax/taxcategories/index.html [hunk 35]
```diff
@@ -19,17 +19,17 @@
...
-            label: taxRate.name,
+            label: taxRate.name|t('site')|e,
             url: taxRate.cpEditUrl,
         }]) %}
     {% endfor %}
 
     {% set tableData = tableData|merge([{
         id: taxCategory.id,
-        title: taxCategory.name,
+        title: taxCategory.name|t('site'),
```
Comment (angrybrad) [inline src/templates/tax/taxcategories/index.html]: `|e`?

File: src/templates/tax/taxrates/index.html [hunk 36]
```diff
@@ -20,13 +20,13 @@
 {% for taxRate in taxRates %}
     {% set tableData = tableData|merge([{
         id: taxRate.id,
-        title: taxRate.name,
+        title: taxRate.name|t('site'),
```
Comment (angrybrad) [inline src/templates/tax/taxrates/index.html]: `|e`?

File: src/templates/tax/taxrates/index.html [hunk 37]
```diff
@@ -36,12 +36,12 @@
     { name: 'rate', title: Craft.t('commerce', 'Rate') },
     { name: 'included', title: Craft.t('commerce', 'Include in Price?'), callback: function(value) {
       if (value) {
-          return '<span data-icon="check" title="'+Craft.t('app', 'Yes')+'"></span>';
+          return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/tax/taxrates/index.html]: not needed

File: src/templates/tax/taxrates/index.html [hunk 38]
```diff
@@ -36,12 +36,12 @@
...
       }
     } },
     { name: 'vat', title: Craft.t('commerce', 'Is VAT?'), callback: function(value) {
         if (value) {
-            return '<span data-icon="check" title="'+Craft.t('app', 'Yes')+'"></span>';
+            return '<span data-icon="check" title="'+Craft.escapeHtml(Craft.t('app', 'Yes'))+'"></span>';
```
Comment (angrybrad) [inline src/templates/tax/taxrates/index.html]: not needed

File: src/templates/tax/taxzones/index.html [hunk 39]
```diff
@@ -22,16 +22,16 @@
...
-            label: taxRate.name,
+            label: taxRate.name|t('site')|e,
             url: taxRate.cpEditUrl,
         }]) %}
     {% endfor %}
 
 	{% set tableData = tableData|merge([{
         id: taxZone.id,
-        title: taxZone.name,
+        title: taxZone.name|t('site'),
```
Comment (angrybrad) [inline src/templates/tax/taxzones/index.html]: `|e`?

File: src/controllers/CustomersController.php (hunk 1 above)
Comment (nfourtythree) [reply src/controllers/CustomersController.php]: Fixed with 9a8c556c898dcdce1d3fa24f0a5814b31d33d3fa

File: src/templates/promotions/discounts/index.html (hunk 9 above)
Comment (nfourtythree) [reply src/templates/promotions/discounts/index.html]: The title column data is escaped in the Vue app.

The use of `|t('site')` mimics the way Craft does it for things like field names as I assumed this was protocol?

File: src/templates/store-settings/countries/index.html (hunk 30 above)
Comment (nfourtythree) [reply src/templates/store-settings/countries/index.html]: Yes the user can change this in the CP

File: src/templates/store-settings/paymentcurrencies/index.html (hunk 32 above)
Comment (nfourtythree) [reply src/templates/store-settings/paymentcurrencies/index.html]: Auto escaped in the title column

**Pull request merged**